site/src/pages/tutorials/tutorial-blog-rest-step3.mdx

```diff
@@ -0,0 +1,168 @@
+
+By completing this step, you'll learn to map your service with the HTTP POST (<type://@Post>) method and make your own request converter (<type://@RequestConverter>).
+
+## What you need
+
+You need the BlogPost.java file, created from [step 2](/tutorials/tutorial-blog-rest-step2). You can always [download](https://github.com/line/armeria-examples/blob/master/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogPost.java) the full version of this file, instead of creating one yourself.
+
+## 0. Create a service file
+
+Create a service file, BlogService.java. Import required files as you go along the tutorial or copy at once from the [sample file](https://github.com/line/armeria-examples/blob/master/rest-api-annotated-service/src/main/java/example/armeria/server/blog/BlogService.java).
+  ```java filename=BlogService.java
+  public final class BlogService {
+    // Used for serializing and deserializing JSON
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final Map<Integer, BlogPost> blogPosts = new ConcurrentHashMap<>();
+    ...
+  ```
+
+## 1. Map HTTP method
+
+Let's start with [mapping the HTTP POST method](/docs/server-annotated-service#mapping-http-service-methods) with our service method:
+
+1. Declare a service method, `createBlogPost()`, in the class `BlogService`.
+2. Map this service method with the HTTP POST method by adding the <type://@Post> annotations.
+3. Assign the endpoint as `/blogs`.
+
+```java filename=BlogService.java highlight=1
+@Post("/blogs")
+public HttpResponse createBlogPost(BlogPost blogPost) throws JsonProcessingException {}
+```
+
+## 2. Handle parameters
+
+Let's take post information through a request body. Armeria's [request converter](/docs/server-annotated-service#converting-an-http-request-to-a-java-object) converts request parameters in HTTP messages into Java objects for you. In the request converter we define what keys of a JSON object to map with what properties of a Java object.
+
+To handle parameters received for creating a blog post, use <type://RequestConvert>, in these steps. The order may be changed.
+
+- [Write a request converter](#write-a-request-converter)
+- [Assign a request converter](#assign-a-request-converter)
+
+### Write a request converter
+
+Armeria's [request converter](/docs/server-annotated-service#converting-an-http-request-to-a-java-object) converts a request body from a client into a Java object for you.
+
+We can use Armeria's default request converter as is, but here let's give a go at customizing a request converter for our blog post requests. We want to convert blog post details into a Java object.
+
+1. Create a BlogPostRequestConverter.java file and declare a class, implementing the <type://RequestConverterFunction> interface. For simplicity sake, generate impromptu IDs for this tutorial.
```

Comment:
```suggestion
1. Create a BlogPostRequestConverter.java file and declare a class, implementing the <type://RequestConverterFunction> interface. For simplicity's sake, generate impromptu IDs for this tutorial.
```

Reply:
Simplicity's sake doesn't sound grammatically correct to me, somehow.
Changed it to "For the sake of simplicity". Hope this is alright. -> c917dd347bbbe0469e0549a35fbeb3a9bba4df60